swift/Tests/Blockchains/BitconCashTests.swift

```diff
@@ -12,8 +12,8 @@ class BitcoinCashTests: XCTestCase {
     func testExtendedKeys() {
         let wallet = testWallet
 
-        let xprv = wallet.getExtendedPrivateKey(for: .bip44, coin: .bitcoinCash, version: .xprv)
-        let xpub = wallet.getExtendedPubKey(for: .bip44, coin: .bitcoinCash, version: .xpub)
+        let xprv = wallet.getExtendedPrivateKey(curve: .secp256k1, purpose: .bip44, coin: .bitcoinCash, version: .xprv)
+        let xpub = wallet.getExtendedPubKey(curve: .secp256k1, purpose: .bip44, coin: .bitcoinCash, version: .xpub)
```

Comment:
should we have `switch` in the wallet which will return curve for specific blockchain?

Reply:
So deduce the curve from the coin type?

Reply:
yes, as discussed: `curveForCoinType()`, I assume this could be applied for a sign functionality as well

Reply:
I'll do this on a separate PR